From ec563c6346f171689d91bedbe815788f10b66b27 Mon Sep 17 00:00:00 2001
From: Colin Walters <walters@verbum.org>
Date: Fri, 6 Jul 2012 13:33:48 -0400
Subject: [PATCH] build: Add --disable-documentation option

Documentation is a source of many cyclical dependency loops; allow
builders to avoid them for a first pass.  Also, some embedders may not
want man pages on their devices.

I want this patch for gnome-ostree because for now I don't have the
Docbook schemas due to the fact they're only distributed in zip file
format and don't have sane build systems.
---
 configure.ac        |   12 +++++++++---
 doc/man/Makefile.am |    6 +++++-
 2 files changed, 14 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index cceb65a..c022d06 100644
--- a/configure.ac
+++ b/configure.ac
@@ -53,9 +53,15 @@ CC_CHECK_CFLAGS_APPEND([                                        \
   -Wno-missing-field-initializers -Wno-unused-parameter         \
   -fno-common -Wno-switch-enum])
 
-AC_PATH_PROG([XSLTPROC], [xsltproc])
-if test -z "$XSLTPROC"; then
-  AC_MSG_ERROR([xsltproc is needed])
+AC_ARG_ENABLE([documentation],
+	AS_HELP_STRING([--enable-documentation], [enable man pages and HTML]),
+	[], [enable_documentation=yes])
+AM_CONDITIONAL(ENABLE_DOCUMENTATION, test x$enable_documentation = xyes)
+if test x$enable_documentation = xyes; then
+  AC_PATH_PROG([XSLTPROC], [xsltproc])
+  if test -z "$XSLTPROC"; then
+    AC_MSG_ERROR([xsltproc is needed for documentation])
+  fi
 fi
 
 GTK_DOC_CHECK([1.3])
diff --git a/doc/man/Makefile.am b/doc/man/Makefile.am
index 5e4946d..a03418f 100644
--- a/doc/man/Makefile.am
+++ b/doc/man/Makefile.am
@@ -1,11 +1,15 @@
 
 NULL =
 
-man_MANS = 				\
+man_MANS =
+
+if ENABLE_DOCUMENTATION
+man_MANS +=				\
 	udisksctl.1			\
 	udisksd.8			\
 	udisks.8			\
 	$(NULL)
+endif
 
 udisksctl.1 : udisksctl.xml
 	$(XSLTPROC) -nonet http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl $<
-- 
1.7.10.4

